src/runtime/button-api.js

```diff
@@ -122,4 +121,51 @@ export class ButtonApi {
     button.addEventListener('click', callback);
     return button;
   }
+
+  /**
+   *
+   * @param {!../api/subscriptions.ButtonOptions|function():null} optionsOrCallback
+   * @return {!../api/subscriptions.ButtonOptions}
+   * @private
+   */
+  getOptions_(optionsOrCallback) {
+    const options = /** @type {?../api/subscriptions.ButtonOptions} */
+        (typeof optionsOrCallback != 'function' ?
+        optionsOrCallback : null);
+    return options || {'theme': Theme.LIGHT, 'lang': 'en'};
+  }
+
+  /**
+   *
+   * @param {?../api/subscriptions.ButtonOptions|function():null} optionsOrCallback
+   * @param {function():null=} opt_callback
+   * @return {function():null|function(Event):boolean}
+   * @private
+   */
+  getCallback_(optionsOrCallback, opt_callback) {
+    const callback = /** @type {function():null|function(Event):boolean} */ (
```

Comment:
Why the variance with return type? Also return type of `null` is not a clear concept.

Reply:
This is to support two different function `attach` (having `addEventListener`) and 'attachSmartButton` (callback).
Without this, linter fails.